config/routes.rb

```diff
@@ -93,13 +93,16 @@
         end
       end
     end
-    resources :teams, except: [:edit, :update, :destroy] do
+    resources :teamsets, only: [:index, :edit, :update] do
       member do
-        patch :dissolve
-      end
-      collection do
-        patch :dissolve_all
+        patch :clone
         patch :randomize
+        patch :dissolve_all
+      end
+      resources :teams, only: [:show] do
```

Comment:
This is the classic "single source of truth" question. And you want a single source of truth. Multiple sources don't help, because if they conflict then what are you going to do?

I can see it potentially helping with debugging, but it's easier to just have better tests.